### PR TITLE
Cleanup function and variable definition order in isotovideo

### PR DESCRIPTION
**Pull request by okurz**
See: https://progress.opensuse.org/issues/81899

### CODE REVIEW

**Review by kalikiana — CHANGES_REQUESTED** [review]

File: isotovideo [hunk 1]
```diff
@@ -83,6 +84,15 @@ session->enable;
 session->enable_subreaper;
 
 my %options;
+my $return_code;
+my $cmd_srv_process;
+my $command_handler;
+my $testprocess;
+my $cmd_srv_fd;
+my $cmd_srv_port;
+my $backend_process;
```
[the remark this answers is not in the document]
Comment (okurz) [reply isotovideo:93]: I am "just" moving the variable declarations to top.

[the remark this answers is not in the document]
Comment (kalikiana) [reply isotovideo:93]: I saw that,and I think we need to move them elsewhere without intermediate steps.

[the remark this answers is not in the document]
Comment (Martchus) [reply isotovideo:93]: I also don't consider moving them to the top as intermediate improvement. When they're not at the top it is at least clear that those variables are only used by the code below. Now the variable's scope is even bigger. Note that these are not *just* variable declarations but also definitions (although they're initialized with `undef`).

[the remark this answers is not in the document]
Comment (okurz) [reply isotovideo:93]: Well, it's my plan to further segment but before moving to other files I first need to understand the code and go step by step.

**Review by perlpunk — APPROVED** [review]

File: isotovideo (hunk 1 above)
[the remark this answers is not in the document]
Comment (perlpunk) [reply isotovideo:92]: As the goal is to move the subs into a module, I am ok with this intermediate step.

[the remark this answers is not in the document]
Comment (kalikiana) [reply isotovideo:92]: Perhaps speaking in code will help more here: #2037

[the remark this answers is not in the document]
Comment (Martchus) [reply isotovideo:92]: I don't see how this PR helps with those variables.

[the remark this answers is not in the document]
Comment (okurz) [reply isotovideo:91]: I extracted functions as far as I could. I hope you agree with me that it does not make sense to define variables mixed in with subroutine definitions. So maybe the complete picture makes more sense now. But if you prefer I can move variables further down mixed in with subroutines.

[the remark this answers is not in the document]
Comment (okurz) [reply isotovideo:90]: My goal is to have unit-testable code which likely means we need to move stuff to a separate, proper module file, which I would do in the next step.

**Comment by okurz** [comment]
I extracted some parts of the commits here into https://github.com/os-autoinst/os-autoinst/pull/2040 which might be easier to accept as a first step.

**Comment by okurz** [comment]
rebased after #2040 was merged

File: isotovideo (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalikiana) [reply isotovideo:90]: I'll re-phrase once more. Currently the code is not well-tested. You're not adding any new tests here. It's not clear to me how you would unit test this code.

[the remark this answers is not in the document]
Comment (kraih) [reply isotovideo:90]: I would make the removal of global variables a priority, that will lead to easier to maintain code.

[the remark this answers is not in the document]
Comment (okurz) [reply isotovideo:90]: ok, so should I move the variables into subroutines and pass them around where necessary?

[the remark this answers is not in the document]
Comment (kraih) [reply isotovideo:90]: Yea, some can probably become object properties later on. Passing them around now will make turning functions into methods quite a bit easier.

[the remark this answers is not in the document]
Comment (Martchus) [reply isotovideo:94]: > I hope you agree with me that it does not make sense to define variables mixed in with subroutine definitions.

Somehow I would actually prefer to have those variables mixed in with subroutine definitions. It still limits the scope of those global variables.

> I would make the removal of global variables a priority, that will lead to easier to maintain code.

I would second that.

**Pull request closed without merging**